alerts/machine-config-operator/MachineConfigControllerDrainError.md

```diff
@@ -0,0 +1,110 @@
+The `machine-config-controller` pod is the central point of management for incoming updates to 
+machines.
+
+For the following command, replace the $CONTROLLERPOD variable with the name of your own `machine-config-controller` pod name. 
+
+```console
+oc -n openshift-machine-config-operator logs $CONTROLLERPOD -c machine-config-controller
+```
+When the MCO starts draining a node, the Machine Config Controller (MCC) records the following log entry:
+
+```console
+1 drain_controller.go:173] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: initiating drain
+```
+The MCC then logs the name of each pod that is drained from the node.
+
+```console
+  1 drain_controller.go:173] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: Evicted pod <namespace>/<pod-name>
+```
+
+If the MCO/MCC is unable to drain a pod after 1m30s, the MCC logs the following error message:
+
+```console
+1 drain_controller.go:173] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: Drain failed. Waiting 1 minute then retrying. Error message from drain: error when waiting for pod "xxxx-xxxx-xx" in namespace "xxxxxxx" to terminate: global timeout reached: 1m30s
+```
+
+If the drain continues to fail, the MCC logs the a second error message:
+
+```console
+ 1 drain_controller.go:173] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: Drain failed. Drain has been failing for more than 10 minutes. Waiting 5 minutes then retrying. Error message from drain: error when waiting for pod "xxxx-xxxx-xx" in namespace "xxxxxxx" to terminate: global timeout reached: 1m30s
+```
+
+After one hour has passed and the drain is still failing, the MCC logs the following error messages and the node is marked degraded.
+
+```console
+1 drain_controller.go:352] node xxxxxxx-xxxxxx-xxxxxx-xxxxx: drain exceeded timeout: 1h0m0s. Will continue to retry.
+```
+
+```console
+1 status.go:126] Degraded Machine: xxxxxxx-xxxxxx-xxxxxx-xxxxx: and Degraded Reason: failed to drain node: xxxxxxx-xxxxxx-xxxxxx-xxxxx after 1 hour. Please see machine-config-controller logs for more information
+```
+
+The MCC logs explicitly list which pods are failing to drain. You should start examining the listed pods for the problem that is causing the drain to fail. Common reasons why a node cannot drain a pod include the following conditions:
+
+- The pod has a PodDisruptionBudget (PBD) that prevents the MCO/MCC from deleting the pod.
+- The pod has storage attached and the kubelet is unable to unmount the storage.
+- The pod has a webhook that is configured to target UPDATE operations or the webhook is not being called by the kube-apiserver.
+- The pod has finalizers set in the pod that are preventing it from terminating.
+
+## Mitigation 
+
+- If a PDB is causing the failure you can temporarily patch it to set the `minAvailable` to 0 so it can scale the pod down successfully. Then patch it to the previous value after the upgrade completes.
+
+```console
+oc patch pdb $PDB -n $NS --type=merge -p '{"spec":{"minAvailable":0}}'
```

Comment:
```suggestion
- If a PDB is causing the failure you can temporarily patch it to set the `minAvailable` to 0 so it can scale the pod down successfully. Then patch it to the previous value after the upgrade completes.
+
```console
oc patch pdb $PDB -n $NS --type=merge -p '{"spec":{"minAvailable":0}}'
```